### DISCUSSION TITLE
Calculate days until next birthday

### DISCUSSION THREAD

**Discussion by AntjeWalter** [neuefische/questions#426] [Web] [answered]
Hi,

I am trying to calculate the days from now until the next birthday of a person. 
The birthday was previously added via an input type="date" and on a dynamically rendered page I want to calculate the age and days until the upcoming birthday of this person. I managed to calculate the age but I can't seem to find any way to use the date input and calculate the days without using the year. Because my previous calculations always gave me a huge number that probably counted the days from the birthday (e.g. in 1992) until now.

Does anybody have an idea about how to only calculate the days?

I pushed my current status [here](https://github.com/AntjeWalter/capstone-project/pull/10) so you can have a look at my [id].js Profile Page and where I want to calculate.

(Also there is a vercel problem for this branch right now...)

**Comment by actyralf** [comment]
You should consider using the [day.js](https://day.js.org/) library. It has a lot of convenient methods for date calculation and manipulation.
Have a look at the "Get + Set" section of the documentation to retrieve information like year, month, day of month etc. With these you should be able to create a date object for the next upcoming birthday. Under the "Durations" section you will find methods to calculate durations (e.g. days between today and the future birthday).

**Reply by AntjeWalter** [reply]
Thank you very much, I will have a look at the library and mark this as answer as soon as I managed to solve the problem :)

**Comment by ufelgen** [comment, marked as answer]
good news: I wrote a function for you, if you're looking for a solution without using a library:

<img width="486" alt="days-until-birthday" src="https://user-images.githubusercontent.com/115288914/209399788-30d54dcf-0191-498f-8804-fbc5949045df.png">

bad news: it's 227 days until my birthday ☹️

**Reply by AntjeWalter** [reply]
Awesome!! It works perfectly fine and is exactly what I needed, thank you 😍
